src/DETHRACE/common/car.c

```diff
@@ -4729,44 +4653,39 @@ void NormalPositionExternalCamera(tCar_spec* c, tU32 pTime) {
     swoop = gCountdown && c->pos.v[1] + 0.001f < gCamera_height;
     manual_swing = gOld_yaw__car != gCamera_yaw || swoop;
     manual_zoom = (double)gOld_zoom != gCamera_zoom;
-    old_camera_pos = *(br_vector3*)&m1->m[3][0];
+    BrVector3Copy(&old_camera_pos, &gCamera->t.t.translate.t);
     if (!gProgram_state.cockpit_on) {
         if (swoop) {
             gCamera_yaw = 0;
             manual_swing = 1;
         }
-        if (fabs(c->speedo_speed) > 0.0006f && gCamera_mode > 0) {
+        if (fabsf(c->speedo_speed) > 0.0006f && gCamera_mode > 0) {
             gCamera_mode = -1;
-            gCamera_sign = m2->m[2][1] * c->direction.v[1]
-                    + m2->m[2][2] * c->direction.v[2]
-                    + m2->m[2][0] * c->direction.v[0]
-                > 0.0;
+            gCamera_sign = BrVector3Dot((br_vector3*)m2->m[2], &c->direction) > 0.0;
         }
         if (c->frame_collision_flag && gCamera_mode != -2) {
             gCamera_mode = 1;
         }
         if (gCar_flying || gCamera_reset || gCamera_mode == -2) {
             gCamera_mode = 0;
         }
-        d = sqrtf((float)gCamera_zoom) + 0.57971013f;
+        d = sqrtf(gCamera_zoom) + 0.57971013f;
```

Comment:
this value is `WORLD_SCALE / 4.f` maybe ?

Reply:
4.f / 6.9f = 0.5797101449275363f
Close enough?

Reply:
```suggestion
        d = sqrtf(gCamera_zoom) + 4.f / WORLD_SCALE;
```